pkg/channel/fanout/fanout_event_handler.go

```diff
@@ -220,13 +219,35 @@ func createEventReceiverFunction(f *FanoutEventHandler) func(context.Context, ch
 			reportArgs.EventType = evnt.Type()
 			reportArgs.Ns = ref.Namespace
 
+			var (
+				httpCount  int = 0
+				httpsCount int = 0
+			)
+
+			for _, sub := range subs {
+				if sub.Subscriber.URL != nil && sub.Subscriber.URL.Scheme == "https" {
+					httpsCount++
+				} else {
+					httpCount++
+				}
+			}
+
+			if httpsCount > 0 {
+				reportArgs.EventScheme = "https"
+			} else {
+				reportArgs.EventScheme = "http"
+			}
```

Comment:
Computing this from the subscribers every time we process an event is an expensive operation. Instead, we should compute this when we set the subscribers on the `FanoutEventHandler`, and store it, possibly as `hasHttpSubs` and `hasHttpsSubs`.